product-service/src/getProductById.js

```diff
@@ -0,0 +1,29 @@
+import productList from './productList.js';
+
+const headers = { "Access-Control-Allow-Origin": "*" };
+
+export const getProductById = async (event) => {
+  console.log('getProductById lambda called with event: ', event);
+  const { productId } = event.pathParameters;
+  // emulate delay from remote URL fetch
+  // suppose we'll get real URL here and not file with hardcoded values
+  await new Promise(resolve => setTimeout(resolve, 2000));
+
+  // search needed product - by id field
+  const product = productList.find((prod) => prod.id === productId);
+
+  if (product === -1){
```

Comment:
Not a big issue, but Array.find returns undefined in case an element was not found. So 404 is not working now

Reply:
Good catch! Fixed, now it's working